### ISSUE TITLE
Confusion about binarization strategy.

### ISSUE THREAD

**Issue by lwmming** [qiuzx2/CIBHash#3]
Thank you for your insightful work! I notice that the "hash_layer", proposed in Greedy Hash, is performed for binarization in this code, which is inconsistent with Eq.11 given in original paper.  Could you please give the reason?

**Comment by zexuanqiu** [comment]
Hi Iwmming, 

In fact, the "hash_layer" just implements the straight-through estimator, in which during the backpropagation function the gradient from the binary code is directly copied to the previous continuous variable, which is consistent in our original paper.
The straight-through estimator is very common in quantization methods.  Indeed, the "hash_layer" code is the same as that of GreedyHash, but that is only a  common method to implement ST estimator via Pytorch.
